src/template.rs

```diff
@@ -46,35 +45,35 @@ impl TryFrom<&str> for Template {
     type Error = TemplateError;
 
     fn try_from(src: &str) -> Result<Self, Self::Error> {
-        let (has_error, is_dynamic) = StrftimeItems::new(src).fold((false, false), |pair, item| {
-            (pair.0 || is_error(&item), pair.1 || is_dynamic(&item))
-        });
-        if has_error {
-            Err(TemplateError::StrftimeError)
-        } else {
-            Ok(Template {
-                src: src.into(),
-                src_bytes: Vec::from(src.as_bytes()).into(),
-                has_ts: is_dynamic,
-                has_fields: RE.is_match(src),
-            })
-        }
+        Template::try_from(src.to_owned())
```

Comment:
This will be useless `to_owned()` if we will get error from `TryFrom<String>`. I think it's a good case for [std::borrow::Cow](https://doc.rust-lang.org/std/borrow/enum.Cow.html): both `TryFrom<String>` and `TryFrom<&str>` can use `TryFrom<Cow<'_, str>>` what should help to avoid extra clones.

Reply:
Updated, thanks!